### PR TITLE
Prep release v0.15.0

### LINKED ISSUE #452: light-client: update sled dependency to latest version
https://github.com/spacejam/sled/releases
https://github.com/informalsystems/tendermint-rs/blob/02fa831b93b3896c7356da07ff81aa0c8852ead7/light-client/Cargo.toml#L21

Can we have something like [dependabot](https://dependabot.com/rust/) to (semi)automate this kind of issues and PRs?

### PR DESCRIPTION
**Pull request by liamsi**
Sorry if this PR grew too big. I can break it into smaller ones if required but I think it should be straightforward to review.
touches parts of: https://github.com/informalsystems/tendermint-rs/issues/388
closes #452
* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Updated CHANGES.md

### CODE REVIEW

File: light-client/Cargo.toml [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
-tendermint-rpc = { version = "0.14.0", path = "../rpc", features = ["client"] }
+tendermint-rpc = { version = "0.15.0", path = "../rpc", features = ["client"] }
 
 anomaly = { version = "0.2.0", features = ["serializer"] }
 contracts = "0.4.0"
 crossbeam-channel = "0.4.2"
 derive_more = "0.99.5"
 futures = "0.3.4"
-prost-amino = "0.5.0"
+prost-amino = "0.6.0"
```
Comment (liamsi) [inline light-client/Cargo.toml:16]: @tarcieri  and I recently updated this. This is only to speed up build times.

File: light-client/Cargo.toml [hunk 2]
```diff
@@ -1,20 +1,20 @@
 [package]
 name = "tendermint-light-client"
-version = "0.14.0"
+version = "0.15.0"
 authors = ["Romain Ruetschi <romain@informal.systems>"]
```
Comment (liamsi) [inline light-client/Cargo.toml:4]: @romac who would you like to add here? @brapse for sure, and who else?

File: light-client/src/fork_detector.rs [hunk 3]
```diff
@@ -52,10 +52,8 @@ pub trait ForkDetector: Send {
 /// the given trusted state, and then:
 ///
 /// - If the verification succeeds, we have a real fork
-/// - If verification fails because of lack of trust,
-///   we have a potential fork.
-/// - If verification fails for any other reason, the
-///   witness is deemed faulty.
+/// - If verification fails because of lack of trust, we have a potential fork.
+/// - If verification fails for any other reason, the witness is deemed faulty.
```
Comment (liamsi) [inline light-client/src/fork_detector.rs:56]: We actually never enforce the rules we have via CI by running fmt via nightly at least there.
See: https://github.com/informalsystems/tendermint-rs/blob/a556d77e2a4b92e6d7d653c8784a281cceca3388/.rustfmt.toml#L1-L12

File: light-node/src/bin/light-node/main.rs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-//! Main entry point for LightNode
+//! Main entry point for light-node
```
Comment (liamsi) [inline light-node/src/bin/light-node/main.rs:1]: I'm not sure why this shows up as a separate crate:
image

**Comment by yihuang** [comment]
How about upgrade `sled` to most recent version `0.33`?

**Comment by liamsi** [comment]
> How about upgrade sled to most recent version 0.33?

It was on my radar (#452) but I didn't try it yet as there might be breaking changes in the sled API. Can try on Monday. Feel free to submit a PR that updates it independently of this one in the meantime.

**Comment by liamsi** [comment]
@yihuang seems fine actually: https://github.com/informalsystems/tendermint-rs/pull/454/commits/62822b4eb8144574bb6843e3b7470820c3f389b1

File: light-node/src/rpc.rs [hunk 5]
```diff
@@ -9,10 +10,10 @@ pub use sealed::{Client, Rpc, Server};
 
 /// Run the given [`Server`] on the given address and blocks until closed.
 ///
-/// n.b. The underlying server has semantics to close on drop. Also does it does not offer any way to
-/// get the underlying Future to await, so we are left with this rather rudimentary way to control
-/// the lifecycle. Should we be interested in a more controlled way to close the server we can
-/// expose a handle in the future.
+/// n.b. The underlying server has semantics to close on drop. Also does it does not offer any way
```
Comment (xla) [inline light-node/src/rpc.rs:13]:
```suggestion
/// n.b. The underlying server has semantics to close on drop. Also it does not offer any way
```

File: CHANGES.md [hunk 6]
```diff
@@ -1,4 +1,9 @@
-## Pending
+## [0.15.0] (2020-07-17)
```
Comment (liamsi) [inline CHANGES.md:1]: We should maybe update the date here. I'll merge this for now and publish the crates this afternoon.

File: light-client/src/fork_detector.rs (hunk 3 above)
Comment (brapse) [reply light-client/src/fork_detector.rs:58]: we should probably change this.

Comment (romac) [reply light-client/src/fork_detector.rs:58]: Yeah let's remove the nightly-only formatting directives.

File: light-client/Cargo.toml (hunk 2 above)
Comment (romac) [reply light-client/Cargo.toml:4]: > $ git shortlog -n -s -- light-client

shows

```
    22  Romain Ruetschi
     6  Alexander Simmerl
     4  Ismail Khoffi
     1  Ethan Buchman
     1  Shivani Joshi
     1  yihuang
```

File: light-client/src/fork_detector.rs (hunk 3 above)
Comment (liamsi) [reply light-client/src/fork_detector.rs:58]: ref: https://github.com/informalsystems/tendermint-rs/issues/153 and https://github.com/informalsystems/tendermint-rs/pull/121

Either remove them or reject PRs via CI that do not comply with these rules.

**Pull request merged**